src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Helpers/Steps/AddPackageStep/AddPackagesStep.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Microsoft.DotNet.Scaffolding.Helpers.General;
+using Microsoft.DotNet.Scaffolding.Helpers.Services;
+
+namespace Microsoft.DotNet.Scaffolding.Helpers.Steps.AddPackageStep;
+
+internal class AddPackagesStep : ScaffoldStep
+{
+    public required IList<string?> PackageNames { get; init; }
```

Comment:
Why do we allow a null package name?

Reply:
nullable strings are nice when we evaluate certain conditions to maybe add a package name, checking for a null before installing is easier and shorter than a bunch conditional ifs to add to this list. If you feel strongly I am not supposed to enforcing a non-nullable string list

Reply:
I get what you're saying, especially since you're in the process of writing the code that adds those package names at the same time you're doing this part 😊 But from an API perspective, thinking about what the AddPackageStep should allow, I don't think it makes sense to allow null strings (to the extent that nullability really disallows it). We'll have to handle the empty set case. But I think the API would be cleaner if we went with non-nullable. The inputs to that should probably be being validated before they're passed to this anyway, I would guess (in the soon-to-be world)? And actually they're non-nullable strings where they're defined as constants now so I think you won't have to do any checking at the moment anyway?

Reply:
fixed, ty